### DISCUSSION TITLE
Followup question to #5415

### DISCUSSION THREAD

**Discussion by kernie** [fwupd/fwupd#5433] [Q&A] [answered]
**Describe the question**
After learning from #5415 I contacted ASROCK support to ask for activation of the CPU support in the BIOS of the motherboard.

I got following back: 

> BIOS department ask if you have more closer information which BIOS option or feature need to be enable for HSI-1 security?

Since I'm now the man-in-the-middle I wonder if you can provide a reference I can refer to?

Thank you!

**Comment by hughsie** [comment]
@superm1 works for AMD, maybe he can share what exact thing the OEM needs to enable in the BIOS build?

**Comment by superm1** [comment, marked as answer]
For completeness sake, please share run the following and share the `dmesg.txt`
```
# dmesg --clear
# rmmod ccp
# modprobe ccp dyndbg
# dmesg > dmesg.txt
```

> @superm1 works for AMD, maybe he can share what exact thing the OEM needs to enable in the BIOS build?

I can't talk about that kind of stuff publicly.  I'd suggest you ask them to contact their "AMD BAE" to discuss the details.  But keep in mind CNY is about to start, so probably won't make progress for a week or two.

**Comment by kernie** [comment]
dmesg.txt:
```
[ 2899.277892] ccp 0000:11:00.2: ccp: unable to access the device: you might be running a broken BIOS.
[ 2899.277894] ccp 0000:11:00.2: psp: unable to access the device: you might be running a broken BIOS.
[ 2902.770316] SVM: TSC scaling supported
[ 2902.770318] kvm: Nested Virtualization enabled
[ 2902.770319] SVM: kvm: Nested Paging enabled
[ 2902.770323] SVM: Virtual VMLOAD VMSAVE supported
[ 2902.770323] SVM: Virtual GIF supported
[ 2902.770323] SVM: LBR virtualization supported
```

**Reply by superm1** [reply]
OK, so you're hitting this: https://github.com/torvalds/linux/blob/a9f94826e4bb6402e67f3eb849dee0811f1de6da/drivers/crypto/ccp/psp-dev.c#L72
Basically the PSP capabilities register isn't populated.  You can add this information to your request to ASRock, and it might help them.

**Reply by kernie** [reply]
So, I let them know. They mentioned CNY, too...

Thank you, @superm1!

**Reply by jvoisin** [reply]
```
[ 1161.694483] ccp 0000:0a:00.1: ccp: unable to access the device: you might be running a broken BIOS.
[ 1161.694504] ccp 0000:0a:00.1: psp does not support SEV
[ 1161.694505] ccp 0000:0a:00.1: psp does not support TEE
[ 1161.694506] ccp 0000:0a:00.1: psp enabled
```

here as well, on a PRIME X570-PRO with a AMD Ryzen 7 3700X, running BIOS version 4408 (2022/11/29)

**Comment by kernie** [comment]
So, ASROCK came back on me and send a new BIOS. I flashed it and got finally the "CPU supported" flag! Moreover, the RAM encryption (TSME) now shows also activated.

From `fwupdmgr get-devices --show-all-devices`:
![cpu_supported](https://user-images.githubusercontent.com/23321751/217031225-8965e5ae-d043-4e8c-aa4f-6b66c672c956.png)

The dmesg result now looks like:
```
[ 1050.472685] ccp 0000:11:00.2: ccp: unable to access the device: you might be running a broken BIOS.
[ 1050.472726] ccp 0000:11:00.2: psp does not support SEV
[ 1050.472731] ccp 0000:11:00.2: tee: ring address: high = 0x2 low = 0x247e0000 size = 32768
[ 1050.482844] ccp 0000:11:00.2: tee enabled
[ 1050.482850] ccp 0000:11:00.2: psp enabled
```

@superm1 I promised ASROCK people to send them feedback. Do you have a recommendation what else is needed for the first message of "broken BIOS"?

**Reply by hughsie** [reply]
Can you ask if ASROCK would consider uploading that to the LVFS? Feel free to give them my contact info.

**Reply by superm1** [reply]
Looks good to me now

**Reply by kernie** [reply]
> Can you ask if ASROCK would consider uploading that to the LVFS? Feel free to give them my contact info.

Well, they answered: “... this LVFS issue is fixed by BIOS patch, we recommend updating BIOS via instant flash or BIOS flashback button ...” . That means probably no...

Is it possible to apply a BIOS upgrade by LVFS? (Sorry for my ignorance)

**Reply by hughsie** [reply]
> Is it possible to apply a BIOS upgrade by LVFS

The LVFS only distributes firmware that the vendor uploaded themselves. You might be able to use `fwupdtool install-blob` if they supply a valid UEFI capsule, but it's not something I'd suggest trying without some way to recover.

**Reply by jvoisin** [reply]
@kernie "this LVFS issue is fixed by BIOS patch": may I ask what specific version fixes this?

**Reply by superm1** [reply]
> @kernie "this LVFS issue is fixed by BIOS patch": may I ask what specific version fixes this?

Different product. Report your issue to your motherboard vendor.

**Reply by jvoisin** [reply]
Arg, you're absolutely right, sorry for the noise!

**Comment by Guara92** [comment]
Still present on `ASRock X670E Steel Legend` with the bios v1.18
```
[    1.092039] ccp 0000:14:00.2: enabling device (0000 -> 0002)
[    1.093239] ccp 0000:14:00.2: ccp: unable to access the device: you might be running a broken BIOS.
[    1.093242] ccp 0000:14:00.2: psp: unable to access the device: you might be running a broken BIOS.
```

@kernie how do you report the issue?

**Comment by kernie** [comment]
I sent via the support site of Asrock.

**Comment by superm1** [comment]
Here is a [patch series](https://lore.kernel.org/linux-crypto/oCZE3H8ZnqlQsatOdb4HH7_0t7E9fsRqVv1QodA3IeUSDCdx9DP1bWsLOVq6T5iGffMjzMequirUJSM4UpcA1y2LZVUvMsffulaArsPy2cs=@hughsie.com/T/#t) for the kernel that enables the feature on the older SoCs.
